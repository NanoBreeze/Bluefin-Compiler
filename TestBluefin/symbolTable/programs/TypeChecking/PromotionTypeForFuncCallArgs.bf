int f(int a) {
	return 1;
}

float g(float a) {
	return 2;
}

void h(int a, int b, float c) {
}

int main() {
	f(5);
	g(6);
	{
		h(1,2,-3);
	}
}