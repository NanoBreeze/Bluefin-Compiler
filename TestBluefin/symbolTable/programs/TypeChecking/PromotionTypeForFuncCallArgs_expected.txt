1 - t_int - t_int
2 - t_int - t_float
5 - t_int - t_int
f(5) - t_int - t_int
6 - t_int - t_float
g(6) - t_float - t_float
1 - t_int - t_int
2 - t_int - t_int
3 - t_int - t_int
-3 - t_int - t_float
h(1,2,-3) - t_void - t_void
